### PR TITLE
[User Model] Moving Location to its own namespace

### PR DESCRIPTION
**Pull request by emawby**
# Description
## One Line Summary
This PR moves Location to its own namespace.

## Details
Public methods for Location code will go through OneSignalLocation. OneSignalLocation.h is now exposed as a public header.

iOS Languages don't have true namespaces but OneSignal.Location will return a class that has location methods on it.

**This PR does NOT move remote params to core so calls to remote params are commented out for now. A separate PR will move remote params and update all code that is using it.** 

### Motivation
The SDK now has multiple namespaces for our different product features. Location is one of those.

### Scope
Location methods.
This adds a new public header for the OneSignal framework. Eventually Location and IAMs will be their own modules.

### OPTIONAL - Other
**OPTIONAL -** Feel free to add any other sections or sub-sections that can explain your PR better.

# Testing
## Unit testing
N/A

## Manual testing
The example app builds and runs and location code can be hit.

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [x] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [ ] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1175)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h [hunk 1]
```diff
@@ -43,18 +42,19 @@ typedef struct os_last_location {
     double horizontalAccuracy;
 } os_last_location;
 
-@interface OneSignalLocation : NSObject
+@protocol OSLocation <NSObject>
+// - Request and track user's location
++ (void)requestPermission;
++ (void)setLocationShared:(BOOL)enable;
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h:48]: Could be `setShared` now

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h [hunk 2]
```diff
@@ -43,18 +42,19 @@ typedef struct os_last_location {
...
++ (BOOL)isLocationShared;
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h:49]: Can be `isShared` now

**Review by jkasten2 — CHANGES_REQUESTED** [review]

**Review by fhboswell — APPROVED** [review]
Looks Good, other than Josh's comments

Are we using [deprecation warnings / update suggestions](https://stackoverflow.com/questions/20122860/add-deprecated-warning-to-a-class-which-redirected-to-another-using-compatibili) with version 5.x.x at all?

**Review by emawby — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1175)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @jennantilla, @jkasten2, and @nan-li)

___
*[`iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h` line 48 at r1](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1175#-NImvVvwOdrllhch4Gm_-r1-48:-NInLFfq8-2fKd9MTyIa:b-896fix) ([raw file](https://github.com/onesignal/onesignal-ios-sdk/blob/a6c50246ba755b5efed8a28849ae3925afa67f36/iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h#L48)):*
<details><summary><i>Previously, jkasten2 (Josh Kasten) wrote…</i></summary><blockquote>

Could be `setShared` now
</blockquote></details>

Done.
___
*[`iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h` line 49 at r1](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1175#-NImvVvwOdrllhch4Gm_-r1-49:-NInLFxm7zjwqNVtAb61:b-896fix) ([raw file](https://github.com/onesignal/onesignal-ios-sdk/blob/a6c50246ba755b5efed8a28849ae3925afa67f36/iOS_SDK/OneSignalSDK/Source/OneSignalLocation.h#L49)):*
<details><summary><i>Previously, jkasten2 (Josh Kasten) wrote…</i></summary><blockquote>

Can be `isShared` now
</blockquote></details>

Done.

**Comment by jkasten2** [comment]
> Looks Good, other than Josh's comments
> 
> Are we using [deprecation warnings / update suggestions](https://stackoverflow.com/questions/20122860/add-deprecated-warning-to-a-class-which-redirected-to-another-using-compatibili) with version 5.x.x at all?

No since we are just dropping the old APIs instead of deprecating. Major release is the only opportunity to drop APIs.

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**